xbmc/guilib/GUIListItem.h

```diff
@@ -172,6 +186,7 @@ class CGUIListItem
   CGUIListItemLayoutPtr m_layout;
   CGUIListItemLayoutPtr m_focusedLayout;
   bool m_bSelected;     // item is selected or not
+  unsigned int m_currentItem; // current item number within container (starting at 1)
```

Comment:
CGUIListItems are mass objects, for instance every PVR EPG event is a CFileItem which is derived from CGUIListItem. One can easily have tens of thousands of those instances in one Kodi process. Thus, we should think twice before adding new members to this class.

Reply:
I had an implementation based on `ListItem.Properties` first - you point is valid, where could I put this new infolabel instead?

Reply:
Would it be possible to ignore the epggrid layout altogether?

Reply:
No not in this case.
The `CGUIListItem` has no idea about grids, panels, lists or whatever.

Reply:
Also tbh I think it's kinda neglectable.
Given an `unsigend int` takes up around 4 byte of memory, it may be more due to padding though, 1mio items would take up addionally 4mb of ram.
I really can't imagine, that this would break any platform.